packages/flutter_tools/lib/src/devfs.dart

```diff
@@ -533,7 +534,8 @@ class DevFS {
         // Make service protocol requests for each.
         dirtyEntries.forEach((Uri deviceUri, DevFSContent content) {
           final Future<Map<String, dynamic>> operation =
-              _operations.writeFile(fsName, deviceUri, content);
+              _operations.writeFile(fsName, deviceUri, content)
+                  .then((dynamic v) => v?.cast<String, dynamic>());
```

Comment:
Same question here

Reply:
same as above, the api returns `Future<dynamic>`.